### ISSUE TITLE
Nuxt Warning from middleware.

### ISSUE THREAD

**Issue by vanling** [becem-gharbi/nuxt-directus#10]
```[nuxt] Calling `useRoute` within middleware may lead to misleading results. Instead, use the (to, from) arguments passed to the middleware to access the new and old routes.```

I think its pointing to https://github.com/becem-gharbi/nuxt-directus/blob/3b386babfa6a222243f7e1d62156cb7faa12403c/src/runtime/composables/useDirectusAuth.ts#L41 called via https://github.com/becem-gharbi/nuxt-directus/blob/3b386babfa6a222243f7e1d62156cb7faa12403c/src/runtime/middleware/auth.ts#L20 

It's just a warning but maybe there is a way to get around it.

**Comment by becem-gharbi** [comment]
When calling `useDirectusAuth` from route middlewares, `useRoute` is also [invoked](https://github.com/becem-gharbi/nuxt-directus/blob/3b386babfa6a222243f7e1d62156cb7faa12403c/src/runtime/composables/useDirectusAuth.ts#L41) at the `useDirectusAuth` scope. This behaviour may lead to bugs in recent `Nuxt` release. I just need to place `useRoute` at the `resetPassword` scope.

```js
  async function resetPassword(password: string): FetchReturnT<any> {
    const route = useRoute();

    return useAsyncData(() =>
      directus.transport.post("/auth/password/reset", {
        password: password,
        token: route.query.token,
      })
    );
  }
```

**Comment by becem-gharbi** [comment]
Fixed on v0.1.4
